compiler/rustc_parse_format/src/lib.rs

```diff
@@ -731,6 +735,13 @@ impl<'a> Parser<'a> {
             }
         } else if self.consume('?') {
             spec.ty = "?";
+            if let Some(&(_, char)) = self.cur.peek() {
+                match char {
+                    '#' => self.suggest_format_reorder(char),
+                    'x' | 'X' => self.suggest_format_remove(char),
```

Comment:
Suggestion: `maybe_suggest_format_parameter_reorder`. In it, we do something like

```rs
fn maybe_suggest_format_parameter_reorder(&mut self, c: char) {
    let Some(&(_, c)) = self.cur.peek() else {
        return;
    };
    let Some(pos) = self.consume_pos(c) else {
        return;
    };

    // We've seen `?`, but the user wrote `#`/`X`/`x` immediately after.
    let span = self.span(pos - 1, pos + 1);
    let replacement = match c {
        '#' => "#?",
        'x' => "x?",
        'X' => "X?",
        _ => return,
    };
    let sugg = Suggestion::ReorderFormat(span, replacement);

    self.errors.insert(0, ParseError {
        description: format!("unknown format identifier '?{}'", c),
        note: Some("if you intended to print `{`, you can escape it using `{{`".to_owned()),
        label: format!("help: the format parameter should be written  as `{replacement}`"),
        span: pos,
        secondary_label: None,
        suggestion: sugg,
    });
}
```